homeassistant/components/vartastorage/sensor.py

```diff
@@ -0,0 +1,221 @@
+                }
+                thisdict[5] = {
+                    "name": "VARTA state text",
+                    "state": result.state_text,
+                    "device_class": "None",
+                    "state_class": "None",
+                    "unit_of_measurement": "",
+                }
+                thisdict[6] = {
+                    "name": "VARTA active power",
+                    "state": result.active_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[7] = {
+                    "name": "VARTA apparent power",
+                    "state": result.apparent_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[8] = {
+                    "name": "VARTA charge power",
+                    "state": result.charge_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[9] = {
+                    "name": "VARTA discharge power",
+                    "state": result.discharge_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[10] = {
+                    "name": "VARTA error code",
+                    "state": result.error_code,
+                    "device_class": "None",
+                    "state_class": "None",
+                    "unit_of_measurement": "",
+                }
+                thisdict[11] = {
+                    "name": "VARTA production power",
+                    "state": result.production_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[12] = {
+                    "name": "VARTA total production power",
+                    "state": result.total_production_power,
+                    "device_class": "energy",
+                    "state_class": "total_increasing",
+                    "unit_of_measurement": "kWh",
+                }
+                thisdict[13] = {
+                    "name": "VARTA total charged energy",
+                    "state": result.total_charged_energy,
+                    "device_class": "energy",
+                    "state_class": "total",
+                    "unit_of_measurement": "kWh",
+                }
+
+                return thisdict
+        except Exception as failed_update:
+            raise UpdateFailed("Error communicating with API") from failed_update
+
+    coordinator = DataUpdateCoordinator(
```

Comment:
Move the coordinator into the `__init__.py`

Reply:
Hey Frenck,
could you give me some background for this change request? As i forwarded the setup from the __init__.py to the sensor.py i thought it would be the right place to place the coordinator.

Of course i will adapt the code according to the needs but i am not very sure how. Do i need to pass the coordinator object as well from the __init__.py to the sensor.py?

Reply:
When done in the `__init__.py` the coordinator can abort the setup of the integration and hint Home Assistant to automatically try again (e.g., when a temporary connection issue occurs).

Additionally, if one is adding other entity platforms in the future, that would be easier as well.

Reply:
Fully understood now - thank you. I will implement all recommended changes and commit again if i am done.

Reply:
moved the coordinator into __init__.py